src/Module.js

```diff
@@ -17,6 +17,42 @@ import ModuleScope from './ast/scopes/ModuleScope.js';
 import { encode } from 'sourcemap-codec';
 import { SourceMapConsumer } from 'source-map';
 
+// Dynamic Import support for acorn
+let moduleDynamicImportsReturnBinding;
+tt._import.startsExpr = true;
+acornPlugins.dynamicImport = ( instance ) => {
+	instance.extend( 'parseStatement', nextMethod => {
+		return function parseStatement ( ...args ) {
+			const node = this.startNode();
+			if ( this.type === tt._import ) {
+				const nextToken = this.input[this.pos];
+				if ( nextToken === tt.parenL.label ) {
+					const expr = this.parseExpression();
+					return this.parseExpressionStatement( node, expr );
+				}
+			}
+			return nextMethod.apply( this, args );
+		}
+	});
+
+	instance.extend( 'parseExprAtom', nextMethod => {
+		return function parseExprAtom ( refDestructuringErrors ) {
+			if ( this.type === tt._import ) {
+				const node = this.startNode();
+				this.next();
+				if ( this.type !== tt.parenL ) {
+					this.unexpected();
+				}
+				if ( moduleDynamicImportsReturnBinding ) {
+					moduleDynamicImportsReturnBinding.push( node );
+				}
+				return this.finishNode( node, 'Import' );
+			}
+			return nextMethod.call( this, refDestructuringErrors );
+		};
+	});
+};
+
```

Comment:
If we could extract the plugin code to a separate file, that would be great. Especially since I expect this code to be removed as soon as acorn rolls out official support. This would also align nicely with one of my current goals to clean up `Bundle.js` and `Module.js` as much as possible 😉

Reply:
Sure I've done this.

Reply:
Btw, I made a PR to check how alive that plugin is: kesne/acorn-dynamic-import#11. In case it's not maintained I would also be willing to fork it.